File: WebApi/Application/Models/CustomJwtPayload.cs
namespace WebApi.Application.Models;

public class CustomJwtPayload
{
    public string NameIdentifier { get; set; }
    public string Role { get; set; }
    public string Hash { get; set; }
    public long? Exp { get; set; }
    public string Iss { get; set; }
    public string Aud { get; set; }
}

File: ReactApp/src/types/CustomJwtPayload.ts
interface CustomJwtPayload {
  "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier"?: string;
  "http://schemas.microsoft.com/ws/2008/06/identity/claims/role"?: "Admin" | "User";
  "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/hash"?: string;
  exp?: number;
  iss?: string;
  aud?: string;
}
